### ISSUE TITLE
Move IsFriendsWith, IsInGroup, and GetRankInGroup out of `retrostudio` library

### ISSUE THREAD

**Issue by break-core** [vopwn55/powerlang-issues#46]
Title is very self explanatory, but if you want more information:

These 3 functions are currently available in the `retrostudio` library; however, they are not very accurate to the use in actual Roblox.

The current usage of the functions is:
```
IsFriendsWith(Player: object, UserId: number) -> Result
IsInGroup(Player: object, GroupId: number) -> Result
GetRankInGroup(Player: object, GroupId: number) -> Result
```

In Roblox, these are attached to a `Player` instance. So for example, you would call it like `somePlayer.IsFriendsWith(261)` instead of `retrostudio.IsFriendsWith(somePlayer, 261)`

TL;DR move these functions as instance functions for `Player`
